plugin/externalPluginScript.go

```diff
@@ -12,10 +12,31 @@ import (
 // externalPluginScript represents an external plugin's script
 type externalPluginScript interface {
 	Path() string
-	InvokeAndWait(ctx context.Context, method string, entry *externalPluginEntry, args ...string) ([]byte, error)
+	InvokeAndWait(ctx context.Context, method string, entry *externalPluginEntry, args ...string) ([]byte, invokeContext, error)
 	NewInvocation(ctx context.Context, method string, entry *externalPluginEntry, args ...string) *internal.Command
 }
 
+type invokeContext struct {
```

Comment:
I think this should be `invocation` and command/stdout/stderr should be their raw types (so `internal.Command` and `[]byte` for the streams). Then the type signatures for the `Invoke*` methods would change to:
```
([]byte, invocation, error)
invocation
```
which reads a bit better.

Reply:
I guess the raw types isn't too important, especially for the streams. For the command object, I was thinking that munging it is easy enough we may as well return it as-is so that it's easier to access its other methods if we ever need to do so in the future.